pytd/writer.py

```diff
@@ -49,10 +63,23 @@ def _cast_dtypes(dataframe, inplace=True, keep_list=False):
             t = "Int64" if df[column].isnull().any() else "int64"
         elif kind == "f":
             t = float
-        elif kind == "O" and keep_list:
-            if df[column].apply(lambda x: isinstance(x, (list, np.ndarray))).all():
+        elif kind == "O":
+            if df[column].apply(_check_column_type, args=[(list, np.ndarray)]).all():
+                t = object
```

Comment:
`t = object` happens in all three conditions. Can we move it to right after `elif kind == "O":`?

```py
elif kind == "O":
    t = object
    if df[column].apply(_check_column_type, args=[(list, np.ndarray)]).all():
```

Reply:
fixed d97e189